src/olympia/api/fields.py

```diff
@@ -108,8 +112,10 @@ def flat(self):
 
     def get_requested_language(self):
         request = self.context.get('request', None)
-        if request and request.method == 'GET' and 'lang' in request.GET:
-            return request.GET['lang']
+        if request and request.method == 'GET' and (lang := request.GET.get('lang')):
+            if not self.flat and not re.fullmatch(LANGUAGE_CODE_REGEX, lang):
+                self.fail('unknown_locale', lang_code=escape(lang))
```

Comment:
Still not sure there is any value in escaping that for HTML. @bobsilverberg what does the frontend typically do with error messages coming from the API ? Are they run through `DOMPurify` ? I see we're returning the same kind of error, with the `lang_code` not escaped at all, when writing translations.

Reply:
🤷‍♂️ given the entire point of the bug was us returning the value as-is in the response I figured it we should at least escape it in the error message.

Reply:
I think we should be consistent and stick to one way of handling parameters in error messages across the API - As I said in the bug, I'm not convinced we care about the escaping at all tbh.

Reply:
Agree on consistency, but the escaping was the only thing that could be seen as "fixing" the bug.  If we're no longer doing that, then this pr is fixing an adjacent issue - validating the locale code (I can file an issue).  I suppose the bug is a wontfix then.

(imo a client is as likely to include a json value in html without sanitizing as a json key)

Reply:
I suppose it doesn't hurt to escape it, but then we should also escape other error messages and not just this one, and we're back to being blocked on what the frontend does to avoid accidentally double-escaping

Reply:
In at least the standard case, where we are using `errorHandler.renderError()` we do not appear to be doing anything at all, other than outputting the message from the error. I can check for other instances where we might be explicitly outputting errors.

Reply:
Looking through the codebase, I am confident that we always render error messages via `renderError()`, which means we are not doing any processing of the message at all.

Reply:
With https://github.com/mozilla/addons-frontend/issues/12040 handling that on our side, I believe we can wontfix the escaping but still do the validation. I personally don't think it's very useful to include the broken locale in the error message, but it shouldn't be escaped IMHO - it's the client's responsibility to deal with content as they see fit.

Reply:
https://github.com/mozilla/addons-server/pull/20438/commits/9173f9f195a01f92c5a28ce08cf92b0184d366b9 drops the sanitizing entirely